pkg/operator/cassandra/controller/controller.go

```diff
@@ -0,0 +1,304 @@
+			return fmt.Errorf("error syncing '%s', requeueing: %s", key, err.Error())
+		}
+		cc.queue.Forget(obj)
+		logger.Infof("Successfully synced '%s'", key)
+		return nil
+	}(obj)
+
+	if err != nil {
+		runtime.HandleError(err)
+		return true
+	}
+
+	return true
+}
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Cluster
+// resource with the current status of the resource.
+func (cc *ClusterController) syncHandler(key string) error {
+
+	// Convert the namespace/name string into a distinct namespace and name.
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	// Get the Cluster resource with this namespace/name
+	cluster, err := cc.clusterLister.Clusters(namespace).Get(name)
+	if err != nil {
+		// The Cluster resource may no longer exist, in which case we stop processing.
+		if apierrors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("cluster '%s' in work queue no longer exists", key))
+			return nil
+		}
+		return err
+	}
+
+	logger.Infof("handling cluster object: %+v", cluster)
+	// Deepcopy here to ensure nobody messes with the cache.
+	return cc.Sync(cluster.DeepCopy())
+}
+
+// enqueueCluster takes a Cluster resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than Cluster.
+func (cc *ClusterController) enqueueCluster(obj *cassandrav1alpha1.Cluster) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	cc.queue.AddRateLimited(key)
+}
+
+// handleObject will take any resource implementing metav1.Object and attempt
+// to find the Cluster resource that 'owns' it. It does this by looking at the
+// objects metadata.ownerReferences field for an appropriate OwnerReference.
+// It then enqueues that Cluster resource to be processed. If the object does not
+// have an appropriate OwnerReference, it will simply be skipped.
+func (cc *ClusterController) handleObject(obj interface{}) {
+	var object metav1.Object
+	var ok bool
+	if object, ok = obj.(metav1.Object); !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			runtime.HandleError(fmt.Errorf("error decoding object, invalid type"))
+			return
+		}
+		object, ok = tombstone.Obj.(metav1.Object)
+		if !ok {
+			runtime.HandleError(fmt.Errorf("error decoding object tombstone, invalid type"))
+			return
+		}
+		logger.Infof("Recovered deleted object '%s' from tombstone", object.GetName())
+	}
+	logger.Infof("Processing object: %s", object.GetName())
+	if ownerRef := metav1.GetControllerOf(object); ownerRef != nil {
+		// TODO: If this object is not owned by a Cluster, we should not do anything more with it.
+		if ownerRef.Kind != "Cluster" {
```

Comment:
should this check be a bit stronger (fully qualified name or include group maybe)?

Reply:
Yes, I feel that this check should indeed be stronger, especially since we have multiple Cluster objects.
Will fix.